protobuf_deps.bzl

```diff
@@ -178,3 +179,12 @@ def protobuf_deps():
         name = "nuget_python_x86-64_3.10.0",
         sha256 = "4474c83c25625d93e772e926f95f4cd398a0abbb52793625fa30f39af3d2cc00",
     )
+
+
+def protobuf_register_toolchains():
+    "registers source toolchain"
+    native.register_toolchains("//:protoc_toolchain")
```

Comment:
I don't think using `:all` is good as we have had problems with in the past. It is implicit and hides a lot of information, like targets being sorted before being registered. also it might register stuff we didn't meant to. 

We usually create two .bzl files that users need to load in their WORKSPACE. one for bringing in the dependencies and then performing toolchain registration. Wouldn't that be preferable instead of a new package?